File: go/pkg/parser/generic_argument_visitor.go
package parser

type GenericArgumentVisitor struct {
	*BaseMojoParserVisitor
}

func NewGenericArgumentVisitor() *GenericArgumentVisitor {
	return &GenericArgumentVisitor{}
}

func (g *GenericArgumentsVisitor) VisitGenericArgument(ctx *GenericArgumentContext) interface{} {
	return nil
}

File: java/mojo-parser/src/main/java/org/mojolang/mojo/parser/GenericArgumentVisitor.java
package org.mojolang.mojo.parser;

import com.google.protobuf.GeneratedMessageV3;
import org.mojolang.mojo.lang.GenericArgument;
import org.mojolang.mojo.lang.NominalType;

public class GenericArgumentVisitor extends MojoBaseVisitor<com.google.protobuf.GeneratedMessageV3> {
    private GenericArgument.Builder builder_ = GenericArgument.newBuilder();

    @Override
    public GeneratedMessageV3 visitGeneric_argument(MojoParser.Generic_argumentContext ctx) {
        NominalTypeVisitor visitor = new NominalTypeVisitor();
        NominalType type = (NominalType)visitor.visitTypeAndAttributes(ctx.type_(), ctx.attributes());

        builder_.setName(type.getName());
        builder_.addAllGenericArguments(type.getGenericArgumentsList());
        builder_.addAllAttributes(type.getAttributesList());

        return builder_.build();
    }
}
